src/transformers/generation/logits_process.py

```diff
@@ -1085,20 +1085,134 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
+
+            # Set the parameters for diverse beam search
+            num_beams = 8  # higher is more diverse
+            num_beam_groups = 4  # 4 groups of 2 beams will explore 4*2=8 beams (=num_beams). by separating the beams into groups and applying penalties within groups, the model is encouraged to explore different sequence possibilities in each group
+            diversity_penalty = 5.5  # enforces diversity among different groups of beams, discourages beams within a group from selecting the same tokens
+
+            # Generate three diverse summaries using the `generate` method
+            outputs_diverse = model.generate(
+                encoder_input_ids,
+                max_length=100,
+                num_beams=num_beams,
+                num_beam_groups=num_beam_groups,
+                diversity_penalty=diversity_penalty,
+                no_repeat_ngram_size=2,
+                early_stopping=True,
+                num_return_sequences=3,
+            )
+
+            # Generate two non-diverse summaries
+            outputs_non_diverse = model.generate(
+                encoder_input_ids,
+                max_length=100,
+                num_beams=num_beams,
+                no_repeat_ngram_size=2,
+                early_stopping=True,
+                num_return_sequences=2,
+            )
+
+            # Decode and print the summaries
+            summaries_diverse = tokenizer.batch_decode(outputs_diverse, skip_special_tokens=True)
+            summaries_non_diverse = tokenizer.batch_decode(outputs_non_diverse, skip_special_tokens=True)
+
+            # Print the results
+            print("Diverse Summaries:")
+            for summary in summaries_diverse:
+                print(summary)
+                # summary 1:  the solar system formed 4.6 billion years ago from the collapse of a giant interstellar molecular cloud
+                # summary 2: the solar system formed 4.6 billion years ago from the collapse of a giant interstellar molecular cloud. of the objects that orbit the Sun directly, the largest are the eight planets, says john mccartney jr.
+                # summary 3: solar system formed 4.6 billion years ago from collapse of interstellar molecular cloud. largest of the eight planets orbit the Sun directly, with the remainder being smaller objects, such as dwarf planet and small solar System bodies - nicolaus mills-simons: the largest are the dwarf worlds and the solar systems' bodies.
+
+            print("\nNon-Diverse Summaries:")
+            for summary in summaries_non_diverse:
+                print(summary)
+                # summary 1:  the solar system formed 4.6 billion years ago from the collapse of a giant interstellar molecular cloud.
+                # summary 2:  the solar system formed 4.6 billion years ago from the collapse of a giant interstellar molecular cloud.
+    ```
+                For more details, see [Diverse Beam Search: Decoding Diverse Solutions from Neural Sequence
```

Comment:
Final nit 2: let's move this to above the example, and decrease the indentation -- it is not rendering correctly (see [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_25481/en/internal/generation_utils#transformers.HammingDiversityLogitsProcessor) the preview)